### PR TITLE
[IMPROVEMENT] Adding release webhook to update version and baseline regression tests

### PR DESCRIPTION
**Pull request by satyammittal**
Please prefix your pull request with one of the following: **[FEATURE]** **[FIX]** **[IMPROVEMENT]**.

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [contributors guide](https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**My familiarity with the project is as follows (check one):**

- [ ] I have never used the project.
- [ ] I have used the project briefly.
- [ ] I have used the project extensively, but have not contributed previously.
- [x] I am an active contributor to the project.

---
#174 
Firstly release number is updated using tagline.
and then regression test table is updated.
References:
[https://developer.github.com/v3/activity/events/types/#releaseevent](https://developer.github.com/v3/activity/events/types/#releaseevent)

### CODE REVIEW

File: mod_ci/controllers.py [hunk 1]
```diff
@@ -568,6 +569,36 @@ def start_ci():
...
+                # Github recommends adding v to the version
+                if release_version[0] == 'v':
+                    release_version = release_version[1:]
+                release_date = payload['created_at']
+                release_commit = GeneralData.query.filter(GeneralData.key == 'last_commit').first()
+                release = CCExtractorVersion(release_data, release_version, release_commit)
+                g.db.add(release)
+                g.db.commit()
+                # adding test corresponding to last commit to the baseline regression results
+                test = Test.query.filter(Test.commit == release_commit).first()
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: We also select on Linux platform, so I'd like to see an additional filter on that.

File: mod_ci/controllers.py [hunk 2]
```diff
@@ -568,6 +569,36 @@ def start_ci():
                 issue.status = issue_data['state']
                 g.db.commit()
 
+        elif event == "release":
+            release_data = payload['release']
+            release_type = payload['prerelease']
+            # checking whether it is meant for production
+            if release_type is False:
+                return
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: I'd like to see a log line indicating this "error" 👍

File: mod_ci/controllers.py [hunk 3]
```diff
@@ -568,6 +569,36 @@ def start_ci():
...
+                g.db.add(release)
+                g.db.commit()
+                # adding test corresponding to last commit to the baseline regression results
+                test = Test.query.filter(Test.commit == release_commit).first()
+                test_result_file = g.db.query(TestResultFile).filter(
+                                    TestResultFile.test_id == test.id).subquery()
+                test_result = g.db.query(TestResult).filter(
+                                    TestResult.test_id == test.id).subquery()
+                g.db.query(RegressionTestOutput.correct).filter(
+                                RegressionTestOutput.regression_id == test_result_file.c.regression_test_id &
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: You'll have to wrap the sections in parenthesis if you want to use `&` according to the docs: http://docs.sqlalchemy.org/en/latest/core/sqlelement.html#sqlalchemy.sql.expression.and_

Comment (satyammittal) [reply mod_ci/controllers.py]: i forgot it

Comment (satyammittal) [reply mod_ci/controllers.py]: i will add

**Pull request merged**